File: client-flutter/lib/models/trainer.dart
class Trainer {
  final String id;
  final String firstName;
  final String lastName;
  final String shortName;
  final String? picture;
  final String? color;

  Trainer({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.shortName,
    this.picture,
    this.color,
  });

  String get fullName => '$firstName $lastName';

  factory Trainer.fromJson(Map<String, dynamic> json) {
    final first = json['firstname']?.toString() ?? '';
    final last = json['lastname']?.toString() ?? '';
    final shortName = json['shortName']?.toString() ??
        '${first.isNotEmpty ? first[0] : ''}${last.isNotEmpty ? last[0] : ''}';

    return Trainer(
      id: json['id']?.toString() ?? '',
      firstName: first,
      lastName: last,
      shortName: shortName,
      picture: json['picture']?.toString(),
      color: json['color']?.toString(),
    );
  }
}

File: client-ios/SihlClient/Models/Trainer.swift
import Foundation

/// Pendant zu `models/trainer.dart`.
struct Trainer: Identifiable {
    let id: String
    let firstName: String
    let lastName: String
    let shortName: String
    let picture: String?
    let color: String?

    var fullName: String { "\(firstName) \(lastName)" }

    init(json: [String: Any]) {
        let first = (json["firstname"] as? String) ?? ""
        let last  = (json["lastname"]  as? String) ?? ""
        self.id        = "\(json["id"] ?? "")"
        self.firstName = first
        self.lastName  = last
        self.shortName = (json["shortName"] as? String)
                      ?? "\(first.first.map(String.init) ?? "")\(last.first.map(String.init) ?? "")"
        self.picture   = json["picture"] as? String
        self.color     = json["color"]   as? String
    }
}
